classes/04class/exercises/c04-iac02/carloshz4/vpc.tf

```diff
@@ -0,0 +1,126 @@
+# Creates VPC
+resource "aws_vpc" "devopsacademy-iac" {
+  cidr_block           = "${var.aws_ip_cidr_range}"
+  enable_dns_support   = true
+  enable_dns_hostnames = true
+  tags = {
+    Name = "devopsacademy-iac"
+  }
+}
+
+
+
+#Creates Internet Gateway
+resource "aws_internet_gateway" "da-igw" {
+  vpc_id = "${aws_vpc.devopsacademy-iac.id}"
+
+  tags = {
+    Name = "devopsacademy-igw-iac"
+  }
+}
+
+
+
+#Creates subnets
+resource "aws_subnet" "subnet1" {
+  cidr_block              = "${var.subnet-cidr["subnet1"]}"
+  vpc_id                  = "${aws_vpc.devopsacademy-iac.id}"
+  availability_zone       = "ap-southeast-2a"
+  map_public_ip_on_launch = true
+  tags = {
+    Name = "${var.subnet-name["subnet1"]}"
+  }
+}
+
+resource "aws_subnet" "subnet2" {
+  cidr_block              = "${var.subnet-cidr["subnet2"]}"
+  vpc_id                  = "${aws_vpc.devopsacademy-iac.id}"
+  availability_zone       = "ap-southeast-2b"
+  map_public_ip_on_launch = true
+  tags = {
+    Name = "${var.subnet-name["subnet2"]}"
+  }
+}
+
+resource "aws_subnet" "subnet3" {
+  cidr_block        = "${var.subnet-cidr["subnet3"]}"
+  vpc_id            = "${aws_vpc.devopsacademy-iac.id}"
+  availability_zone = "ap-southeast-2b"
+  tags = {
+    Name = "${var.subnet-name["subnet3"]}"
+  }
+}
+
+resource "aws_subnet" "subnet4" {
+  cidr_block        = "${var.subnet-cidr["subnet4"]}"
+  vpc_id            = "${aws_vpc.devopsacademy-iac.id}"
```

Comment:
As you already have a `map` in the input, please use `for_each` here to simpler and more flexible code.

https://www.terraform.io/docs/configuration/resources.html#for_each-multiple-resource-instances-defined-by-a-map-or-set-of-strings